app/lib/project/MetapolatorProject.js

```diff
@@ -509,8 +509,54 @@ define([
         return this._controller;
     };
 
-    _p.import = function(masterName, sourceUFODir, glyphs) {
-        var importer = new ImportController( this._log, this,
+    _p.importZippedUFOInstances = function(blob) {
+        var mem_io = new InMemory()
+          , imported_instances = Array()
+          ;
+
+        zipUtil.unpack(false, blob, mem_io, "");
+
+        var dirs = mem_io.readDir(false, "/")
+          , baseDir = dirs[0]
+          , names = mem_io.readDir(false, baseDir)
+          , n, l
+          ;
+
+        console.log("dirs:", dirs);
+
+        for (n=0, l=names.length; n<l; n++){
+            var name = names[n]
+              , UFOZip = baseDir + name //This may be wrong in some cases. We need a more robust implementation.
+              , another_blob
+              ;
+
+            console.log("UFOZip:", UFOZip);
+
+            another_blob = mem_io.readFile(false, UFOZip);
+            zipUtil.unpack(false, another_blob, mem_io, baseDir);
+        }
+
+        names = mem_io.readDir(false, baseDir);
+        for (n=0, l=names.length; n<l; n++){
+            var name = names[n];
+            if (name[name.length-1]=='/'){
+                var sourceUFODir = baseDir + name.split("/")[0] //This may be wrong in some cases. We need a more robust implementation.
```

Comment:
Best put a `FIXME: ` above and describe roughly what a robust implementation would do. It may be enough to describe what you are actually expecting `name` here to be.